src/limiter.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Module dependencies.
+ */
+
+import assert from 'assert';
+import memoryAdapterFactory from './adapters/memory';
+
+/**
+  * Expose `Limiter`.
+  */
+export default class Limiter {
+    /**
+    * Initialize a new limiter with an adapter and `opts`:
+    *
+    *  - `id` identifier being limited
+    *  - `max` maximum number of calls
+    *  - `duration` the duration before reseting the number of calls
+    *
+    * @param {Object} opts
+    * @param {Object} getFn: the function which will be called to get the rate limiting data
+    * @api public
+    */
+    constructor(opts, adapterFactory = memoryAdapterFactory()) {
```

Comment:
we should also support `db` for backward compatibility and write a deprecated message with `console.log` (I hate this, but couldn't think on something else)